### PR TITLE
fix build error introduced in PR #1297 and #1298

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

File: sdk/core/azure-core/CMakeLists.txt [hunk 1]
```diff
@@ -124,7 +124,9 @@ target_include_directories(azure-core PUBLIC ${CURL_INCLUDE_DIRS})
 target_include_directories(azure-core INTERFACE ${nlohmann_json_INCLUDE_DIRS})
 target_link_libraries(azure-core INTERFACE Threads::Threads nlohmann_json::nlohmann_json)
 
-if(NOT MSVC)
+if(MSVC)
+    target_link_libraries(azure-core PRIVATE crypt32)
+else()
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CMakeLists.txt:129]: I don't understand. What was the error?

From what I understood, Azure core doesn't depend on crypt32. Did the base64 implementation depend on that somehow? If so, how?

Comment (Jinming-Hu) [reply sdk/core/azure-core/CMakeLists.txt:129]: Yes, https://docs.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-cryptbinarytostringa

Comment (ahsonkhan) [reply sdk/core/azure-core/CMakeLists.txt:129]: How did this pass CI?

Does linking to `crypt32` give us access to `wincrypt.h`? How are these two related?

Comment (Jinming-Hu) [reply sdk/core/azure-core/CMakeLists.txt:129]: There are many ways to build this project, we can build with cmake, or ninjia, or directly open the project with Visual Studio. and also there are many platforms out there. So passing CI doesn't mean this project can be built on any other platforms. I think we should try our best to be compatible with more platforms.

> Does linking to crypt32 give us access to wincrypt.h?

No, `wincrypt.h` is in the default directory. Like we don't have to do anything to include `windows.h`.

Comment (ahsonkhan) [reply sdk/core/azure-core/CMakeLists.txt:129]: > So passing CI doesn't mean this project can be built on any other platforms. I think we should try our best to be compatible with more platforms.

Gotcha. That makes sense.

> No, `wincrypt.h` is in the default directory. Like we don't have to do anything to include `windows.h`.

Then what is requiring us to link `crypt32`? Can you share the build error you see if we don't have this change? I still don't understand why we need to link `crypt32` in core and what that gets us.

Comment (Jinming-Hu) [reply sdk/core/azure-core/CMakeLists.txt:129]:
```
5>azure-core.lib(base64.obj) : error LNK2019: unresolved external symbol __imp_CryptStringToBinaryA referenced in function "class std::vector<unsigned char,class std::allocator<unsigned char> > __cdecl Azure::Core::Base64Decode(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?Base64Decode@Core@Azure@@YA?AV?$vector@EV?$allocator@E@std@@@std@@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@4@@Z)
5>azure-core.lib(base64.obj) : error LNK2019: unresolved external symbol __imp_CryptBinaryToStringA referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl Azure::Core::Base64Encode(class std::vector<unsigned char,class std::allocator<unsigned char> > const &)" (?Base64Encode@Core@Azure@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEBV?$vector@EV?$allocator@E@std@@@4@@Z)
5>C:\Users\jinmhu.FAREAST\Desktop\azure-sdk-for-cpp\build\sdk\core\azure-core\test\ut\Debug\azure-core-test.exe : fatal error LNK1120: 2 unresolved externals
5>Done building project "azure-core-test.vcxproj" -- FAILED.
```

Comment (Jinming-Hu) [reply sdk/core/azure-core/CMakeLists.txt:129]: basically `wincrypt.h` is for compilation, while `crypt32.lib` is for linking, two different procedures during build.

Speaking of being compatible with more platforms, I did find a link error in core sdk when linking against static libcurl library (`libcurl.a`). We do want to be compatible with both static and dynamic dependency libraries, right?

Comment (ahsonkhan) [reply sdk/core/azure-core/CMakeLists.txt:129]: > We do want to be compatible with both static and dynamic dependency libraries, right?

I suppose so. @antkmsft, @vhvb1989 can you confirm that's a goal for the SDK?

**Pull request merged**